lib/matplotlib/pyplot.py

```diff
@@ -1199,7 +1200,27 @@ def subplot(*args, **kwargs):
 
         # add ax2 to the figure again
         plt.subplot(ax2)
+
+        # make the first axes "current" again
+        plt.subplot(221)
+
     """
+    proj_placeholder = object()
+    # sort out if the user passed us either polar or projection
+    # in principle we need to handle if *axes_class* is passed, but
+    # we will leave that validation to down stream code.  Here we will
+    # only normalize `polar=True` vs `projection='polar'` so we can decide
+    # if we should create a new Axes or return the existing one
+    projection = kwargs.get('projection', proj_placeholder)
+    polar = kwargs.pop('polar', proj_placeholder)
```

Comment:
Is it really OK to modify `kwargs` here? If the user has called `plt.subplot(111, **kwargs)`, then they may be surprised that kwargs is modified.

Reply:
That's a new dict that gets instantiated in the function call.